python/lsst/obs/hsc/transformRegistry.py

```diff
@@ -247,20 +255,75 @@ class HscDistortionConfig(Config):
+    tolerance = Field(dtype=float, default=5.0e-3, doc="Tolerance for inversion (pixels)")  # Much less than 1
+    maxIter = Field(dtype=int, default=10, doc="Maximum iterations for inversion")  # Usually sufficient
     plateScale = Field(dtype=float, default=1.0, doc="Plate scale (arcsec/mm)")
 
 
+def makeAstPolyMapCoeffs(order, xCoeffs, yCoeffs):
+    """Convert polynomial coefficients in HSC format to AST PolyMap format
+
+    Paramaters
+    ----------
+    order: int
+        Polynomial order
+    xCoeffs, yCoeffs: list of float
+        Forward or inverse polynomial coefficients for the x and y axes
+        of output, in this order:
+            x0y0, x0y1, ...x0yN, x1y0, x1y1, ...x1yN-1, ...
+        where N is the polynomial order.
+
+    Returns
+    -------
+    Forward or inverse coefficients for astshim.PolyMap
+    as a 2-d numpy array.
+    """
+    nCoeffs = (order + 1) * (order + 2) // 2
+    if len(xCoeffs) != nCoeffs:
+        raise ValueError("found %s xCcdToSky params; need %s" % (len(xCoeffs), nCoeffs))
+    if len(yCoeffs) != nCoeffs:
+        raise ValueError("found %s yCcdToSky params; need %s" % (len(yCoeffs), nCoeffs))
+
+    coeffs = np.zeros([nCoeffs * 2, 4])
+    i = 0
+    for nx in range(order + 1):
+        for ny in range(order + 1 - nx):
+            coeffs[i] = [xCoeffs[i], 1, nx, ny]
+            coeffs[i + nCoeffs] = [yCoeffs[i], 2, nx, ny]
+            i += 1
+    assert i == nCoeffs
+    return coeffs
+
+
 def makeHscDistortion(config):
-    """ Make an HscDistortion object
-    @param[in] config: pexConfig.Config object containing the distortion parameters needed to construct the
-    transform
+    """Make an HSC distortion transform
+
+    Note that inverse coefficients provided, but they are not accurate enough
+    to use: test_distortion.py reports an error of 2.8 pixels
+    (HSC uses pixels for its focal plane units) when transforming
+    from pupil to focal plane. That explains why the original HSC model uses
+    the inverse coefficients in conjunction with iteration.
+
+    Parameters
+    ----------
+    config: HscDistortionConfig
+        Distortion coefficients
+
+    Returns
+    -------
+    focalPlaneToPupil: lsst.afw.geom.TransformPoint2ToPoint2
+        Transform from focal plane to field angle coordinates
     """
-    skyToCcd = DistortionPolynomial(config.skyToCcdOrder, config.skyToCcdOrder,
-                                    config.xSkyToCcd, config.ySkyToCcd)
-    ccdToSky = DistortionPolynomial(config.ccdToSkyOrder, config.ccdToSkyOrder,
-                                    config.xCcdToSky, config.yCcdToSky)
-    return HscDistortion(skyToCcd, ccdToSky, config.plateScale*arcseconds, config.tolerance, config.maxIter)
+    forwardCoeffs = makeAstPolyMapCoeffs(config.ccdToSkyOrder, config.xCcdToSky, config.yCcdToSky)
+
+    # Note that the actual error can be somewhat larger than TolInverse;
+    # the max error I have seen is less than 2, so I scale conservatively
```

Comment:
Why is there a need to scale anything? Can't you just use the raw number in the config?

Reply:
Because the unit test might fail, as per the comment. I'd rather have the config parameter passed directly to the astshim Mapping, despite it not being a perfect upper limit, rather than try to guess how much to scale it. Guessing here isn't as bad because the test either passes or it doesn't (I added more margin than needed here because the exact amount of error isn't interesting).